vector/src/Data/Vector/Primitive.hs

```diff
@@ -201,14 +201,9 @@ type role Vector nominal
 -- representationally equal type. The operation just changes the type of the
 -- underlying pointer and does not modify the elements.
 --
--- This is marginally safer than 'unsafeCast', since this function imposes an
--- extra 'Coercible' constraint. This function is still not safe, however,
--- since it cannot guarantee that the two types have memory-compatible
--- 'Prim' instances.
---
```

Comment:
How come this haddock is removed? Thanks to your previous PR that added that comment made me realize that `unsafeCast` was not implemented for primitive vector, so I added it in #401

Reply:
That part was there by accident. I copied it from the docs for `Data.Vector.Storable.unsafeCoerceVector` to adjust it for this function, but then decided to adjust the original doc comment and forgot to remove this part. The part below it says essentially the same (though we can alternatively keep this part and delete the part below it, if you like this better).

Reply:
It is a valuable comment, so please bring it back

Reply:
So you mean bring this part back and remove the part below it? Because they essentially say the same...

Reply:
You can remove the part below or consolidate the two paragraphs into one if you want, but the important parts are:

> `This is marginally safer than 'unsafeCast', since this function imposes an extra 'Coercible' constraint.`

> `The @Coercible@ constraint guarantees that the element types are representationally equal. It however cannot guarantee that their respective 'Prim'  instances are compatible.`